flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistryImpl.java

```diff
@@ -239,7 +239,15 @@ public void shutdown() {
 
 			if (queryService != null) {
 				stopTimeout = new FiniteDuration(1L, TimeUnit.SECONDS);
-				stopFuture = Patterns.gracefulStop(queryService, stopTimeout);
+
+				try {
+					stopFuture = Patterns.gracefulStop(queryService, stopTimeout);
+				} catch (IllegalStateException ignored) {
+					// this can happen if the underlying actor system has been stopped before shutting
+					// the metric registry down
+					// TODO: Pull the MetricQueryService actor out of the MetricRegistry
+					LOG.debug("Cannot gracefully stop the metric query service actor.");
```

Comment:
good point

Reply:
Giving it a second look, I deliberately did not include the exception, because it can only happen if the `ActorSystem` has been shut down before. I'll change the debug log message instead.